buildSrc/src/test/java/org/elasticsearch/gradle/precommit/ForbiddenPatternsTaskTests.java

```diff
@@ -0,0 +1,127 @@
+package org.elasticsearch.gradle.precommit;
+
+import org.elasticsearch.gradle.test.GradleUnitTestCase;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class ForbiddenPatternsTaskTests extends GradleUnitTestCase {
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    public void testCheckInvalidPatternsWhenNoSourceFilesExist() throws Exception {
+        Project project = createProject();
+        ForbiddenPatternsTask task = createTask(project);
+
+        task.checkInvalidPatterns();
+
+        File outputMarker = new File(project.getBuildDir(), "markers/forbiddenPatterns");
+        Optional<String> result = Files.readAllLines(outputMarker.toPath(), StandardCharsets.UTF_8).stream().findFirst();
+
+        assertTrue(result.isPresent());
+        assertEquals("done", result.get());
+    }
+
+    public void testCheckInvalidPatternsWhenSourceFilesExistNoViolation() throws Exception {
+        Project project = createProject();
+        ForbiddenPatternsTask task = createTask(project);
+
+        String line = "public void bar() {}";
+        File file = new File(project.getProjectDir(), "src/main/java/Foo.java");
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+        Files.write(file.toPath(), line.getBytes());
+
+        task.checkInvalidPatterns();
+
+        File outputMarker = new File(project.getBuildDir(), "markers/forbiddenPatterns");
+        Optional<String> result = Files.readAllLines(outputMarker.toPath(), StandardCharsets.UTF_8).stream().findFirst();
+
+        assertTrue(result.isPresent());
+        assertEquals("done", result.get());
+    }
+
+    public void testCheckInvalidPatternsWhenSourceFilesExistHavingTab() throws Exception {
+        Project project = createProject();
+        ForbiddenPatternsTask task = createTask(project);
+
+        String line = "\tpublic void bar() {}";
```

Comment:
I would extract a `writeSourceFile(name, String... lines)` method and use it in each test.